api/src/main/java/org/eclipse/microprofile/reactive/messaging/connector/IncomingConnectorFactory.java

```diff
@@ -0,0 +1,108 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.reactive.messaging.connector;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.reactive.messaging.MessagingProvider;
+import org.eclipse.microprofile.reactive.streams.operators.PublisherBuilder;
+
+import java.util.NoSuchElementException;
+
+/**
+ * SPI used to implement a connector managing a source of messages for a specific <em>transport</em>. Typically, to
+ * handle the consumption of records from Kafka, the reactive messaging extension would need to implement a {@code bean}
+ * implementing this interface. This bean is called for every {@code stream} that need to be created for this specific
+ * <em>transport</em> (so Kafka in the example). These streams are connected to methods annotated with
+ * {@link org.eclipse.microprofile.reactive.messaging.Incoming}.
+ * <p>
+ * The factory is called to create a {@code stream} for each configured <em>transport</em>. The configuration is done using
+ * MicroProfile Config. The following snippet gives an example for a hypothetical Kafka connector:
+ *
+ * <pre>
+ * mp.messaging.incoming.my-stream.type=i.e.m.reactive.messaging.impl.kafka.Kafka
+ * mp.messaging.incoming.my-stream.bootstrap-servers=localhost:9092
+ * mp.messaging.incoming.my-stream.topic=my-topic
+ * ...
+ * </pre>
+ * <p>
+ * This configuration keys are structured as follows: {@code mp.messaging.[incoming|outgoing].stream-name.attribute}.
+ * <p>
+ * The {@code stream-name} segment in the configuration key corresponds to the name of the stream used in the
+ * {@code Incoming} annotation:
+ *
+ * <pre>
+ * &#64;Incoming("my-stream")
+ * public void consume(String s) {
+ *      // ...
+ * }
+ * </pre>
+ * <p>
+ * The set of attributes depend on the connector and transport layer (typically, bootstrap-servers is Kafka specific). The
+ * {@code type} attribute is mandatory and indicates the fully qualified name of the {@link MessagingProvider}
+ * implementation. It must match the class returned by the {@link #type()} method. This is how a reactive messaging
+ * implementation looks for the specific {@link IncomingConnectorFactory} required for a stream. In the previous
+ * configuration, the reactive extension implementation would need to find the {@link IncomingConnectorFactory} returning
+ * the {@code i.e.m.reactive.messaging.impl.kafka.Kafka} class as result to its {@link #type()} method to create the
+ * {@code my-stream} stream. Note that if the connector cannot be found, the deployment must be failed.
```

Comment:
Do you want to say " with a DeploymentException" ?

Reply:
Done, in several places.